geonode/base/models.py

```diff
@@ -540,6 +558,21 @@ class ResourceBase(PolymorphicModel, PermissionLevelMixin, ItemBase):
     """
     Base Resource Object loosely based on ISO 19115:2003
     """
+    BASE_PERMISSIONS = {
```

Comment:
Better to define this dictionary outside, similar to the other permissions.

This might be useful in general.

Reply:
My code assumes it is defined on a class level. My code assumes a polymorphic mechanism according to these dictionaries. More to PERMISSIONS dict but BASE_PERMISSIONS are here for consistency

Reply:
ok, so forget my comment